### PR TITLE
Cli OPT for surface elevation

### PR DESCRIPTION
**Pull request by akshaysridhar**
modified:   driver.jl
	modified:   sphere/baroclinic_wave_utilities.jl
	new file:   sphere/topography.jl
	modified:   types.jl

# PULL REQUEST

## Purpose and Content
Towards coupler interface, with surface elevation supported.

## Benefits and Risks
Risks: Can be mitigated by ensuring interface preserves existing test results in "no topographical elevation" scenario.

## Linked Issues
(Provide references to any link issues. Use closes #issuenum to automatically close an open issue)
- #408
 
## PR Checklist
- [x] This PR has a corresponding issue OR is linked to an SDI.
- [x] I have followed CliMA's codebase [contribution](https://clima.github.io/ClimateMachine.jl/latest/Contributing/) and [style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines OR N/A.
- [x] I have followed CliMA's [documentation policy](https://github.com/CliMA/policies/wiki/Documentation-Policy).
- [x] I have checked all issues and PRs and I certify that this PR does not duplicate an open PR.
- [x] I linted my code on my local machine prior to submission OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code used in an integration test OR N/A. (Buildkite)
- [x] All tests ran successfully on my local machine OR N/A.
- [x] All classes, modules, and function contain docstrings OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: examples/hybrid/driver.jl [hunk 1]
```diff
@@ -213,11 +216,18 @@ include("../common_spaces.jl")
 
 include(joinpath("sphere", "baroclinic_wave_utilities.jl"))
 
+if topography == "DCMIP200"
+    warp_function = topography_dcmip200
+elseif topography == "NoWarp"
+    warp_function = nothing
+end
+
```
Comment (charleskawczynski) [inline examples/hybrid/driver.jl:224]: Let's move this into `get_spaces` to clean up global scope.

File: examples/hybrid/driver.jl [hunk 2]
```diff
@@ -29,6 +29,8 @@ zd_rayleigh = parsed_args["zd_rayleigh"]
 zd_viscous = parsed_args["zd_viscous"]
 κ₂_sponge = parsed_args["kappa_2_sponge"]
 t_end = FT(time_to_seconds(parsed_args["t_end"]))
+dt_save_to_sol = time_to_seconds(parsed_args["dt_save_to_sol"])
+topography = parsed_args["topography"]
```
Comment (charleskawczynski) [inline examples/hybrid/driver.jl:33]: Let's move this into `get_spaces` to clean up global scope.

File: examples/hybrid/driver.jl [hunk 3]
```diff
@@ -39,6 +41,7 @@ t_end = FT(time_to_seconds(parsed_args["t_end"]))
 @assert parsed_args["config"] in ("sphere", "column")
 @assert rayleigh_sponge in (true, false)
 @assert viscous_sponge in (true, false)
+@assert topography in ("NoWarp", "Earth", "DCMIP200")
```
Comment (charleskawczynski) [inline examples/hybrid/driver.jl:44]: Let's move this into `get_spaces` to clean up global scope.

File: examples/hybrid/driver.jl [hunk 4]
```diff
@@ -29,6 +29,8 @@ zd_rayleigh = parsed_args["zd_rayleigh"]
 zd_viscous = parsed_args["zd_viscous"]
 κ₂_sponge = parsed_args["kappa_2_sponge"]
 t_end = FT(time_to_seconds(parsed_args["t_end"]))
+dt_save_to_sol = time_to_seconds(parsed_args["dt_save_to_sol"])
```
Comment (charleskawczynski) [inline examples/hybrid/driver.jl:32]:
```suggestion
```

File: .buildkite/pipeline.yml [hunk 5]
```diff
@@ -225,6 +225,22 @@ steps:
         command: "julia --color=yes --project=examples examples/hybrid/driver.jl --vert_diff true --moist equil --rad clearsky --microphy 0M --job_id sphere_aquaplanet_rhoe_equilmoist_clearsky --dt 200secs --t_end 2.5days"
         artifact_paths: "sphere_aquaplanet_rhoe_equilmoist_clearsky/*"
 
+      - label: ":computer: held suarez (ρe_tot) topography"
+        command: "julia --color=yes --project=examples examples/hybrid/driver.jl --forcing held_suarez --job_id sphere_held_suarez_rhoe_dcmip_topography --t_end 6days --topography DCMIP200 --dt 200secs --regression_test false --upwinding third_order --rayleigh_sponge true --viscous_sponge true"
```
Comment (jiahe23) [inline .buildkite/pipeline.yml:229]: do we have to use both sponges here?

Comment (akshaysridhar) [reply .buildkite/pipeline.yml:229]: No; I added an item in buildkite to demonstrate that

File: .buildkite/longruns/pipeline.yml [hunk 6]
```diff
@@ -92,6 +92,19 @@ steps:
...
+        command: "julia --color=yes --project=examples --threads=8 examples/hybrid/driver.jl --forcing held_suarez --upwinding third_order --dt 300secs --t_end 1200days --fps 30 --job_id longrun_hs_rhoe_third --dt_save_to_sol 1days --dt_save_to_disk 10days --topography DCMIP200"
+        artifact_paths: "longrun_hs_rhoe_third/*"
+        agents:
+          slurm_cpus_per_task: 8
+      
+      - label: ":computer: held suarez (ρe_tot) third order topography"
+        command: "julia --color=yes --project=examples --threads=8 examples/hybrid/driver.jl --forcing held_suarez --upwinding third_order --dt 300secs --t_end 1200days --fps 30 --job_id longrun_hs_rhoe_third --dt_save_to_sol 1days --dt_save_to_disk 10days --topography DCMIP200"
+        artifact_paths: "longrun_hs_rhoe_third/*"
+        agents:
+          slurm_cpus_per_task: 8
```
Comment (jiahe23) [inline .buildkite/longruns/pipeline.yml:106]: THe same simulation is added twice.

Comment (jiahe23) [reply .buildkite/longruns/pipeline.yml:106]: Ah, and for longrun, I think we will also need sponge?

File: examples/hybrid/driver.jl [hunk 7]
```diff
@@ -213,6 +213,7 @@ include("../common_spaces.jl")
 
 include(joinpath("sphere", "baroclinic_wave_utilities.jl"))
 
+
```
Comment (jiahe23) [inline examples/hybrid/driver.jl:216]: not necessary

File: examples/hybrid/driver.jl (hunk 4 above)
Comment (charleskawczynski) [reply examples/hybrid/driver.jl:32]: cc @akshaysridhar?

File: examples/hybrid/driver.jl (hunk 2 above)
Comment (charleskawczynski) [reply examples/hybrid/driver.jl:33]: cc @akshaysridhar

File: .buildkite/longruns/pipeline.yml (hunk 6 above)
Comment (charleskawczynski) [reply .buildkite/longruns/pipeline.yml:106]: cc @akshaysridhar

**Comment by charleskawczynski** [comment]
Looks like there's unresolved comments

**Comment by akshaysridhar** [comment]
@charleskawczynski Resolved the two remaining comments (Thanks for catching this, I think the rebase closed those conversations so I didn't see them while scanning the PR comments).

**Review by charleskawczynski — APPROVED** [review]

**Pull request merged**